presto-orc/src/test/java/com/facebook/presto/orc/TestFilterFunction.java

```diff
@@ -0,0 +1,143 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.orc;
+
+import com.facebook.presto.spi.ConnectorSession;
+import com.facebook.presto.spi.Page;
+import com.facebook.presto.spi.block.Block;
+import com.facebook.presto.spi.block.DictionaryBlock;
+import com.facebook.presto.spi.block.LongArrayBlock;
+import com.facebook.presto.spi.relation.Predicate;
+import com.facebook.presto.testing.TestingConnectorSession;
+import com.google.common.collect.ImmutableList;
+import org.testng.annotations.Test;
+
+import java.util.Arrays;
+import java.util.Optional;
+
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertTrue;
+
+public class TestFilterFunction
+{
+    private static final long UNLUCKY = 13;
+
+    @Test
+    public void testFilter()
+    {
+        ConnectorSession session = new TestingConnectorSession(ImmutableList.of());
+        FilterFunction filter = new FilterFunction(session, true, new IsOddPredicate());
+
+        Block numbers = makeNumbers(0, 1000);
+        int[] allPositions = makePositions(0, 1000, 1);
+        assertFilter(filter, numbers, allPositions, allPositions.length);
+
+        Block dictionaryNumbers = new DictionaryBlock(numbers, allPositions);
+        // Sparse coverage of the dictionary values
+        int[] sparsePositions = makePositions(1, 300, 3);
+        assertFilter(filter, dictionaryNumbers, sparsePositions, sparsePositions.length);
+
+        // Full coverage of the dictionary values
+        assertFilter(filter, dictionaryNumbers, allPositions, allPositions.length);
+
+        // Test with a different DictionaryBlock over the same numbers. Results are reused. The DictionaryBlock covers the
+        // values sparsely. TheDictionaryBlock itself is accessed sparsely.
+        DictionaryBlock otherDictionary = new DictionaryBlock(numbers, makePositions(1, 332, 3));
+        int[] otherDictionaryPositions = makePositions(0, 150, 2);
+        assertFilter(filter, otherDictionary, otherDictionaryPositions, otherDictionaryPositions.length);
+
+        // Repeat test on a DictionaryBlock over different content to make sure that cached results are not reused.
+        Block numbers2 = makeNumbers(1, 1001);
+        Block dictionary2Numbers = new DictionaryBlock(numbers2, allPositions);
+        assertFilter(filter, dictionary2Numbers, allPositions, allPositions.length);
```

Comment:
don't use a1, a2, a3 names; here, I'd inline numbers2 and dictionary2Numbers to avoid naming problem: `assertFilter(filter, new DictionaryBlock(makeNumbers(1, 1001), allPositions), allPositions, allPositions.length);`